common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -4880,6 +4880,22 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
     LLAP_EXTERNAL_CLIENT_USE_HYBRID_CALENDAR("hive.llap.external.client.use.hybrid.calendar",
         false,
         "Whether to use hybrid calendar for parsing of data/timestamps."),
+
+    // confs for llap-external-client cloud deployment
+    LLAP_EXTERNAL_CLIENT_CLOUD_RPC_PORT("hive.llap.external.client.cloud.rpc.port", 30004,
+        "The LLAP daemon RPC port for external clients when llap is running in cloud environment."),
+    LLAP_EXTERNAL_CLIENT_CLOUD_OUTPUT_SERVICE_PORT("hive.llap.external.client.cloud.output.service.port", 30005,
+                "LLAP output service port when llap is running in cloud environment"),
+    LLAP_EXTERNAL_CLIENT_CLOUD_JWT_SHARED_SECRET_PROVIDER(
+        "hive.llap.external.client.cloud.jwt.shared.secret.provider",
+        "org.apache.hadoop.hive.llap.security.ConfBasedJwtSharedSecretProvider",
```

Comment:
I would also provide environment variable based secret provider. Conf based ones have the potential of plain text secret if jceks is not used. Also will be good to have a config to enable the ports explicitly. If the config is false, then none of these ports, jwt secret etc. should be initialized.

Reply:
It's now changed to DefaultJwtSharedSecretProvider which reads from conf.getPassword, if not found, then tries ENV variable.